### PR TITLE
9742: Default welcome message returns after being deleted #9742

### PR DESCRIPTION
**Pull request by RomaKis**
Default welcome message returns after being deleted #9742

### Description
Column value in core_config_data was nullable. When we saved '' as column was nullable in lib/internal/Magento/Framework/DB/Adapter/Pdo/Mysql.php line 3001 the value became null, so it was null in db. If value is null it will take the default value, so we couldn't save '' and see '' at the form.
Now column value isn't nullable, so when we save '' it will save in db as '' and show at the form as ''.

### Fixed Issues (if relevant)

1. magento/magento2#9742:Default welcome message returns after being deleted

### Manual testing scenarios
1. Go to Content -> Configuration -> Click Edit on theme being used.
2. Open header tab -> clear Welcome Text.
3. Click save and continue.
4. Open header tab -> Welcome Text field should be empty.

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by sidolov** [comment]
Hi @RomaKis, there is no bug with configuration persistence, you can save empty welcome message, field is empty in db (screenshot added).
17607381ee

Issue caused by incorrect form data population (fallback to default value) in \Magento\Theme\Model\Design\Config\DataProvider\MetadataLoader::getData() method. 
The same behavior with Copyright field. 

On frontend both fields are empty
94a239e4a4

**Comment by RomaKis** [comment]
Hi @sidolov, the bug is that they are not empty in backend. When you tried to delete it and save. It will reappear after save. 
You are right that it is getting data from \Magento\Theme\Model\Design\Config\DataProvider\MetadataLoader::getData(), but the problem is deeper. 
The problem is that if we save (for example) Footer -> Copyright  = '' (as empty line) it will be converted as null and saved to db. As Magento see that value is null, so it show the default copyright from app/code/Magento/Theme/etc/config.xml. So it is the same: if Magento has clean installation - it is null or we save it as '' it is null too.

**Comment by sidolov** [comment]
@RomaKis after clean Magento installation records about welcome message and copyrights are absent on db. They will be created if saved values were different then default. It's expected result that empty values stored in db as null, doesn't make sense store empty strings, we don't consider this to be a good design. Described problem exists only on backend form, frontend works as expected, so, issue should be fixed in form data population flow (for example system configuration show empty values correctly).

**Comment by RomaKis** [comment]
@sidolov, ok, I agree with you. I change code. Please, review it again.

File: app/code/Magento/Theme/Model/Design/Config/Storage.php [hunk 1]
```diff
@@ -87,10 +87,14 @@ public function load($scope, $scopeId)
                 $scopeId,
                 $fieldData->getFieldConfig()
             );
-            if ($value !== null) {
-                $fieldData->setValue($value);
+
+            if ($value == null) {
```
Comment (sidolov) [inline app/code/Magento/Theme/Model/Design/Config/Storage.php]: Use strict comparison

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Theme/Model/Design/Config/Storage.php (hunk 1 above)
Comment (RomaKis) [reply app/code/Magento/Theme/Model/Design/Config/Storage.php]: @sidolov, done.

File: app/code/Magento/Theme/Model/Design/Config/Storage.php [hunk 2]
```diff
@@ -87,10 +87,14 @@ public function load($scope, $scopeId)
                 $scopeId,
                 $fieldData->getFieldConfig()
             );
-            if ($value !== null) {
-                $fieldData->setValue($value);
+
+            if ($value === null) {
+                $value = '';
+
```
Comment (sidolov) [inline app/code/Magento/Theme/Model/Design/Config/Storage.php]: Remove blank line at end of control structure

**Review by sidolov — CHANGES_REQUESTED** [review]

File: app/code/Magento/Theme/Model/Design/Config/Storage.php (hunk 2 above)
Comment (RomaKis) [reply app/code/Magento/Theme/Model/Design/Config/Storage.php]: @sidolov, done and squashed.

**Review by sidolov — APPROVED** [review]

**Pull request merged**